keras/src/backend/mlx/numpy.py

```diff
@@ -579,21 +579,33 @@ def maximum(x1, x2):
     return mx.maximum(x1, x2)
 
 
-def median(x, axis=-1, keepdims=False):
-    x = convert_to_tensor(x)
-    x_sorted = mx.sort(x, axis=axis)
-    axis_size = x_sorted.shape[axis]
-    medians = mx.take(
-        x_sorted, indices=mx.array([(axis_size // 2) - 1]), axis=axis
-    )
-    if not keepdims:
-        medians = mx.squeeze(medians, axis=axis)
-    return medians
+def median(x, axis=None, keepdims=False):
```

Comment:
I think this implementation will not support a non-scalar axis or negative axes. Can you check if this was tested? The previous implementation should support it (from reading the code).